@ghellwig @davidlange6

for ( const auto &pDigi: pDigiEB)

was my first choice, but it does not compile:

>> Compiling  /afs/cern.ch/work/a/argiro/ecalpedPCL/CMSSW_9_1_0_pre2/src/Calibration/EcalCalibAlgos/src/ECALpedestalPCLworker.cc 
/afs/cern.ch/work/a/argiro/ecalpedPCL/CMSSW_9_1_0_pre2/src/Calibration/EcalCalibAlgos/src/ECALpedestalPCLworker.cc: In member function 'virtual void ECALpedestalPCLworker::analyze(const edm::Event&, const edm::EventSetup&)':
/afs/cern.ch/work/a/argiro/ecalpedPCL/CMSSW_9_1_0_pre2/src/Calibration/EcalCalibAlgos/src/ECALpedestalPCLworker.cc:38:28: error: 'begin' was not declared in this scope
     for (const auto& pDigi:pDigiEB){


I think something is ill with the underlying digi classes 





> On 11 Apr 2017, at 11:33, Gregor Mittag <notifications@github.com> wrote:
> 
> @ghellwig commented on this pull request.
> 
> In Calibration/EcalCalibAlgos/src/ECALpedestalPCLworker.cc:
> 
> > +
> +// ------------ method called for each event  ------------
> +void
> +ECALpedestalPCLworker::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
> +{
> +    using namespace edm;
> +      
> +    Handle<EBDigiCollection> pDigiEB;
> +    iEvent.getByLabel("ecalDigis","ebDigis",pDigiEB);
> +   
> +    Handle<EEDigiCollection> pDigiEE;
> +    iEvent.getByLabel("ecalDigis","eeDigis",pDigiEE);
> +
> +
> +
> +    for (EBDigiCollection::const_iterator pDigi=pDigiEB->begin(); pDigi!=pDigiEB->end(); ++pDigi){
> 
> or const auto& to preserve the current behaviour
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

--- a/Calibration/EcalCalibAlgos/src/ECALpedestalPCLHarvester.cc
+++ b/Calibration/EcalCalibAlgos/src/ECALpedestalPCLHarvester.cc
@@ -31,11 +31,10 @@ void ECALpedestalPCLHarvester::dqmEndJob(DQMStore::IBooker& ibooker_, DQMStore::
     // calculate pedestals and fill db record
     EcalPedestals pedestals;
 
-    std::cout << "Filling ped record " << std::endl;
-       
+    std::stringstream hname;
     for (uint16_t i =0; i< EBDetId::kSizeForDenseIndexing; ++i) {
         std::stringstream hname;
-        hname<<"EcalCalibration/EcalPedestalPCL/";
+        hname.str("EcalCalibration/EcalPedestalPCL/");
         hname<<"eb_"<<i;
         MonitorElement* ch= igetter_.get(hname.str());
         double mean = ch->getMean();
@@ -64,9 +63,10 @@ void ECALpedestalPCLHarvester::dqmEndJob(DQMStore::IBooker& ibooker_, DQMStore::
         pedestals.setValue(id.rawId(),ped);
     }
 
+
     for (uint16_t i =0; i< EEDetId::kSizeForDenseIndexing; ++i) {
-        std::stringstream hname;
-        hname<<"EcalCalibration/EcalPedestalPCL/";
+
+        hname.str("EcalCalibration/EcalPedestalPCL/");
         hname<<"ee_"<<i;
         MonitorElement* ch= igetter_.get(hname.str());
         double mean = ch->getMean();
@@ -94,7 +94,7 @@ void ECALpedestalPCLHarvester::dqmEndJob(DQMStore::IBooker& ibooker_, DQMStore::
     }
 
 
-    std::cout << "Writing to DB file ... " << std::endl;
+
 
     // write out pedestal record
     edm::Service<cond::service::PoolDBOutputService> poolDbService;

